master/buildbot/changes/github.py

```diff
@@ -97,16 +97,13 @@ def reconfigService(self,
         self._http = yield httpclientservice.HTTPClientService.getService(
             self.master, baseURL, headers=http_headers)
 
-        if not branches:
-            branches = ['master']
-
         self.token = token
         self.owner = owner
         self.repo = repo
         self.branches = branches
         self.project = project
         self.pollInterval = pollInterval
-        self.repository_link = link_urls[repository_link]
+        self.repository_type = link_urls[repository_type]
```

Comment:
The hook does not have this feature.
It is possible that it is not compatible with the reporter which could only know about http repository type (to be verified)